crates/rpc/engine/fork_choice.rs

```diff
@@ -37,16 +37,56 @@ impl RpcHandler for ForkChoiceUpdatedV3 {
             _ => return Err(RpcErr::Internal),
         };
 
-        storage.update_finalized_block_number(finalized_block_number)?;
-        storage.update_safe_block_number(safe_block_number)?;
-        serde_json::to_value(json!({
-            "payloadId": null,
-            "payloadStatus": {
-                "latestValidHash": null,
-                "status": "SYNCING",
-                "validationError": null
-            }
-        }))
-        .map_err(|_| RpcErr::Internal)
+        // Check if the payload is available for the new head hash.
+        let header = match storage.get_block_header_by_hash(self.fork_choice_state.head_block_hash)
+        {
+            Ok(Some(header)) => header, // DO stuff,
+            Ok(None) => return syncing_response(),
+            _ => return Err(RpcErr::Internal),
+        };
+
+        // We are still under the assumption that the blocks are only added if they are connected
+        // to the canonical chain. That means that for the state to be consistent we only need to
+        // check that the safe and finalized ones are in the canonical chain and that the heads parent is too.
+
+        let head_valid = storage
+            .is_canonical(header.number - 1, header.parent_hash)
+            .map_err(|_| RpcErr::Internal)?;
+
+        let safe_valid = storage
+            .is_canonical(safe_block_number, self.fork_choice_state.safe_block_hash)
+            .map_err(|_| RpcErr::Internal)?;
+
+        let finalized_valid = storage
+            .is_canonical(
```

Comment:
instead of adding a new function to the storage (to avoid increasing the surface area of storage), we could add a helper function that calls `storage.get_canonical_block_hash` and compares results